src/jit/lir.cpp

```diff
@@ -1040,8 +1168,14 @@ LIR::Range LIR::AsRange(GenTree* firstNode, GenTree* lastNode)
 //
 // Return Value: The newly constructed range.
 //
-LIR::Range LIR::SetTreeSeq(Compiler* compiler, GenTree* tree)
+// static
+LIR::Range LIR::SeqTree(Compiler* compiler, GenTree* tree)
 {
+    compiler->gtSetEvalOrder(tree);
```

Comment:
> That is, were the cases not calling gtSetEvalOrder on a newly constructed tree doing so on purpose or because they forgot, and it didn't matter?

I think that in most cases it doesn't matter. AFAICT `gtSetEvalOrder` is not really required for correctness when targeting the RyuJIT backend.